### PR TITLE
Fixing button state with error closes #4229

### LINKED ISSUE #4229: When showing the self-help page, the link in the top right corner is not changing to "Take me back to the old look"
image

### PR DESCRIPTION
**Pull request by myleshorton**

### CODE REVIEW

File: src/github.com/getlantern/lantern-ui/app/partials/error.html [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
-        <div><a ng-click="toggleShowXunleiHelp()">{{ 'WHAT_DO' | translate }}</a></div>
-        <div class="error-action ng-hide" ng-show="showXunleiHelp">
-          {{ 'XUNLEI_HELP' | translate }}
-        </div>
-      </li>
-      <li>
-        <div>{{ 'ANOTHER_ISSUE' | translate }}</div>
-        <div><a ng-click="toggleShowConnectionHelp()">{{ 'WHAT_DO' | translate }}</a></div>
+<div ng-controller="ErrorCtrl">
+  <div id="error" ng-show="showError && !showMobileAd && !showBookmarks">
```
Comment (myleshorton) [inline src/github.com/getlantern/lantern-ui/app/partials/error.html]: This is a whitespace change except for the above two lines. These just separate the ng-controller div from the inner div (seems to be a good practice to separate the ng-show logic, for example, from the logic of the controller) but more importantly just adds the !showBookmarks line here to make sure we don't show the bookmarks if we're showing the error.

**Comment by myleshorton** [comment]
For anyone reviewing this, it's definitely a little messy with all the various states. This code is all going to be ported to react soon, though, so probably not worth getting it pristine.

File: src/github.com/getlantern/lantern-ui/app/partials/error.html (hunk 1 above)
Comment (uaalto) [inline src/github.com/getlantern/lantern-ui/app/partials/error.html]: Thanks for the comment, it certainly saved time :)

File: src/github.com/getlantern/lantern-ui/app/index.html [hunk 2]
```diff
@@ -91,10 +91,10 @@
     </div>
     <div id="feedLink" ng-class="{subtle: showNews}">
       <div>
-        <a ng-click="showNewsfeed()" ng-show="!showNews">
+        <a ng-click="showNewsfeed()" ng-show="!showNews && !showError">
           {{ 'SHOW_NEWSFEED' | translate }}
         </a>
-        <a ng-click="hideNewsfeed()" ng-show="showNews">
+        <a ng-click="hideNewsfeed()" ng-show="showNews || showError">
```
Comment (uaalto) [inline src/github.com/getlantern/lantern-ui/app/index.html]: Ok, apart from what we talked in Slack, this seems to address particularly the case of the issue.

**Comment by uaalto** [comment]
LGTM! Great bug fixing spree @myleshorton  :)

**Comment by myleshorton** [comment]
Thanks @uaalto!

**Pull request merged**